x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/utils/NativeMemoryCalculatorTests.java

```diff
@@ -131,26 +132,14 @@ public void testAllowedBytesForMlWhenAutoIsTrue() {
             Settings settings = newSettings(percent, true, ByteSizeValue.ofMb(randomIntBetween(0, 5000)));
             ClusterSettings clusterSettings = newClusterSettings(percent, true, ByteSizeValue.ofMb(randomIntBetween(0, 5000)));
 
-            double truePercent = Math.min(90, ((nodeSize - jvmSize - ByteSizeValue.ofMb(200).getBytes()) / (double) nodeSize) * 100.0D);
-            long expected = Math.round(nodeSize * (truePercent / 100.0));
+            long expected = Math.min(nodeSize - jvmSize - ByteSizeValue.ofMb(200).getBytes(), nodeSize * 9 / 10);
 
             assertThat(NativeMemoryCalculator.allowedBytesForMl(node, settings).getAsLong(), equalTo(expected));
             assertThat(NativeMemoryCalculator.allowedBytesForMl(node, clusterSettings).getAsLong(), equalTo(expected));
             assertThat(NativeMemoryCalculator.allowedBytesForMl(node, percent, true).getAsLong(), equalTo(expected));
         }
     }
 
-    public void testAllowedBytesForMlWhenBothJVMAndNodeSizeAreUnknown() {
-        int percent = randomIntBetween(5, 200);
-        DiscoveryNode node = newNode(null, null);
-        Settings settings = newSettings(percent, randomBoolean(), ByteSizeValue.ofMb(randomIntBetween(0, 5000)));
-        ClusterSettings clusterSettings = newClusterSettings(percent, randomBoolean(), ByteSizeValue.ofMb(randomIntBetween(0, 5000)));
-
-        assertThat(NativeMemoryCalculator.allowedBytesForMl(node, settings), equalTo(OptionalLong.empty()));
-        assertThat(NativeMemoryCalculator.allowedBytesForMl(node, clusterSettings), equalTo(OptionalLong.empty()));
-        assertThat(NativeMemoryCalculator.allowedBytesForMl(node, percent, randomBoolean()), equalTo(OptionalLong.empty()));
-    }
```

Comment:
This test is deleted because in 8.x it should be impossible that we don't know the JVM size for an ML node.  We add the attribute ourselves and have done since before 7.last, so an 8.x node will never be in a cluster with a node that didn't set it.

This also allows some assertions to be added and a `Long` changed to `long` which reduces the risk of subtle bugs being accidentally introduced.